### ISSUE TITLE
Class 'ActionScheduler_ActionFactory' not found

### ISSUE THREAD

**Issue by shahariaazam** [woocommerce/action-scheduler#521]
I used composer to install action-scheduler. Here is an important part of my `composer.json`.
```json
{
  "require": {
    "woocommerce/action-scheduler": "3.1.4"
  },
  "autoload": {
    "classmap": ["lib/"]
  }
}
```
Included the main `action-scheduler.php` file in my plugin file like -

```php
// Loading action-scheduler library
require_once( plugin_dir_path( __FILE__ ) . 'vendor/woocommerce/action-scheduler/action-scheduler.php' );

// Load vendor autoloader
require __DIR__ . '/vendor/autoload.php';
```

On `deactivate` of my plugin. I wanted to clear all existing schedule that produced from my plugin that I identified with group.

```
as_unschedule_all_actions( '', array(), bamiw_get_action_scheduler_group() );
```

So when I tried to deactivate my plugin, it record a fatal error in my log file. Here is the trace details.

```
[12-Apr-2020 23:16:11 UTC] PHP Fatal error:  Uncaught Error: Class 'ActionScheduler_ActionFactory' not found in /var/www/wordpress2/wp-content/plugins/bamiw/vendor/woocommerce/action-scheduler/classes/abstracts/ActionScheduler.php:17
Stack trace:
#0 /var/www/wordpress2/wp-content/plugins/bamiw/vendor/woocommerce/action-scheduler/functions.php(30): ActionScheduler::factory()
#1 /var/www/wordpress2/wp-content/plugins/bamiw/vendor/woocommerce/action-scheduler/classes/migration/Scheduler.php(89): as_schedule_single_action()
#2 /var/www/wordpress2/wp-content/plugins/bamiw/vendor/woocommerce/action-scheduler/classes/migration/Controller.php(99): Action_Scheduler\Migration\Scheduler->schedule_migration()
#3 /var/www/wordpress2/wp-includes/class-wp-hook.php(286): Action_Scheduler\Migration\Controller->schedule_migration()
#4 /var/www/wordpress2/wp-includes/class-wp-hook.php(312): WP_Hook->apply_filters()
#5 /var/www/wordpress2/wp-includes/plugin.php(478): WP_Hook->do_action()
#6 /var/www/wordpress2/wp-includes/load.ph in /var/www/wordpress2/wp-content/plugins/bamiw/vendor/woocommerce/action-scheduler/classes/abstracts/ActionScheduler.php on line 17
[12-Apr-2020 23:16:11 UTC] PHP Stack trace:
[12-Apr-2020 23:16:11 UTC] PHP   1. {main}() /var/www/wordpress2/wp-admin/admin-ajax.php:0
[12-Apr-2020 23:16:11 UTC] PHP   2. do_action() /var/www/wordpress2/wp-admin/admin-ajax.php:175
[12-Apr-2020 23:16:11 UTC] PHP   3. WP_Hook->do_action() /var/www/wordpress2/wp-includes/plugin.php:478
[12-Apr-2020 23:16:11 UTC] PHP   4. WP_Hook->apply_filters() /var/www/wordpress2/wp-includes/class-wp-hook.php:312
[12-Apr-2020 23:16:11 UTC] PHP   5. wp_ajax_delete_plugin() /var/www/wordpress2/wp-includes/class-wp-hook.php:288
[12-Apr-2020 23:16:11 UTC] PHP   6. wp_send_json_success() /var/www/wordpress2/wp-admin/includes/ajax-actions.php:4551
[12-Apr-2020 23:16:11 UTC] PHP   7. wp_send_json() /var/www/wordpress2/wp-includes/functions.php:3999
[12-Apr-2020 23:16:11 UTC] PHP   8. wp_die() /var/www/wordpress2/wp-includes/functions.php:3975
[12-Apr-2020 23:16:11 UTC] PHP   9. _ajax_wp_die_handler() /var/www/wordpress2/wp-includes/functions.php:3291
```

Is there any idea how I can make all classes available anywhere in my plugin?

**Comment by shahariaazam** [comment]
Ooops. It's very weird that if `woocommerce` plugin installed on the site. I don't get this error at all. All works fine.

So somehow besides using this as a standalone package, does this have any dependency with Woocommerce?

**Comment by lkraav** [comment]
It doesn't, but loading order may change.

Did you run `composer dump-autoload`?

**Comment by shahariaazam** [comment]
@lkraav AFAIK I am using 3.1.4 that is not loading classes via composer.
So dump-autoload may not work.

**Comment by rrennick** [comment]
@shahariaazam You should not be enabling Composer autoloading with Action Scheduler. AS has its own internal autoloading functionality which accounts for multiple instances of AS being active at the same time. Does this problem still occur if you remove the Composer autoloading?

**Comment by shahariaazam** [comment]
> @shahariaazam You should not be enabling Composer autoloading with Action Scheduler. AS has its own internal autoloading functionality which accounts for multiple instances of AS being active at the same time. Does this problem still occur if you remove the Composer autoloading?

If you see my example I shared.. I was not loading AS with composer. See my above example.

**Comment by rrennick** [comment]
Can you check your plugin folder for https://github.com/woocommerce/action-scheduler/blob/master/classes/ActionScheduler_ActionFactory.php ?

**Comment by shahariaazam** [comment]
@rrennick yes the file s there.

I install the AS 3.1.4 via composer (but not using it via autoloader). So that class is present in the correct location.

**Comment by rrennick** [comment]
@shahariaazam I'm not sure what to tell you. I have a few test plugins I've created that pull in AS via composer and don't have an issue with any of them. WooCommerce also uses Composer to include AS.
